test/e2e/wait_util.go

```diff
@@ -232,3 +237,82 @@ func WaitForSecret(secretName, secretNamespace string) {
 	})
 	require.NoError(t, err)
 }
+
+// WaitForHTTPResponse waits for an HTTP valid response
+func WaitForHTTPResponse(httpClient http.Client, method, url string, response interface{}) (err error) {
+	logrus.Infof("calling url: %s", url)
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return err
+	}
+
+	eofCount := 0
+	malFormedErrors := 0
+	unexpectedResponseErrors := 0
+	maximunErrors := 5
+
+	err = wait.Poll(retryInterval, timeout*2, func() (done bool, err error) {
+		res, err := httpClient.Do(req)
+
+		if res != nil {
+			logrus.Infof("response status code: %d", res.StatusCode)
+		}
+
+		if err != nil {
+			if strings.Contains(err.Error(), "Timeout exceeded") {
```

Comment:
I think there is a better way to do this, checking the error type
You can do:

```
if e,ok := err.(net.Error); ok && e.Timeout() {
    // This was a timeout
} else if err != nil {
    // This was an error, but not a timeout
}
```
https://pkg.go.dev/net#Error

Reply:
Oh, really interesting. I'll apply this. Thanks!